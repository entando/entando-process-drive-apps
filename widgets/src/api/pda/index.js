import makeRequest from 'api/makeRequest';
import { SERVICE, METHODS } from 'api/constants';

import MOCK_CONNECTIONS from 'mocks/pda/connections';
import MOCK_PROCESSES from 'mocks/pda/processes';

export const getConnections = async () =>
  makeRequest({
    domain: SERVICE.URL,
    uri: '/connections',
    method: METHODS.GET,
    mockResponse: MOCK_CONNECTIONS,
    useAuthentication: false,
  });

export const getProcess = async connection =>
  makeRequest({
    domain: SERVICE.URL,
    uri: `/connections/${connection}/processes/definitions`,
    method: METHODS.GET,
    mockResponse: MOCK_PROCESSES,
    useAuthentication: false,
  });
