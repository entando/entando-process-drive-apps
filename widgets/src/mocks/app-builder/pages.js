const taskListConfigs = {
  payload: {
    code: 'pda_task_list',
    config: {
      groups:
        '[{"label":"Administrators","key":"Administrators","checked":true},{"label":"broker","key":"broker","checked":true},{"label":"admin","key":"admin","checked":false},{"label":"rest-all","key":"rest-all","checked":true},{"label":"kiemgmt","key":"kiemgmt","checked":false},{"label":"kie-server","key":"kie-server","checked":true},{"label":"appraiser","key":"appraiser","checked":false},{"label":"manager","key":"manager","checked":true},{"label":"supplier","key":"supplier","checked":true},{"label":"IT","key":"IT","checked":true},{"label":"PM","key":"PM","checked":true},{"label":"approver","key":"approver","checked":true}]',
      options:
        '[{"key":"newPageOnClick","label":"Open new page on table row click","checked":false},{"key":"showClaim","label":"Show Claim Button","checked":true},{"key":"showComplete","label":"Show Complete Button","checked":true}]',
      columns:
        '[{"name":"id","position":0,"isVisible":true},{"name":"name","position":1,"isVisible":true},{"name":"createdBy","position":2,"isVisible":true},{"name":"createdAt","position":3,"isVisible":true},{"name":"dueTo","position":4,"isVisible":true},{"name":"status","position":5,"isVisible":true},{"name":"description","position":6,"isVisible":true},{"name":"owner","position":7,"isVisible":true},{"name":"inputData","position":8,"isVisible":true},{"name":"outputData","position":9,"isVisible":true},{"name":"priority","position":10,"isVisible":true},{"name":"subject","position":11,"isVisible":true},{"name":"type","position":12,"isVisible":true},{"name":"form","position":13,"isVisible":true},{"name":"activatedAt","position":14,"isVisible":true},{"name":"skipable","position":15,"isVisible":true},{"name":"workItemId","position":16,"isVisible":true},{"name":"processId","position":17,"isVisible":true},{"name":"processDefinitionId","position":18,"isVisible":true},{"name":"parentId","position":19,"isVisible":true},{"name":"slaCompliance","position":20,"isVisible":true},{"name":"slaDueTo","position":21,"isVisible":true},{"name":"potentialOwners","position":22,"isVisible":true},{"name":"excludedOwners","position":23,"isVisible":true},{"name":"businessAdmins","position":24,"isVisible":true}]',
      knowledgeSource: 'kieStaging',
      process: 'Mortgage_Process.MortgageApprovalProcess@mortgage-process_1.0.1-SNAPSHOT',
    },
  },
  errors: {},
};

const taskDetailsConfigs = {
  payload: {
    code: 'phase_1_widgets_task_details',
    config: {
      containerId: 'evaluation_1.0.0-SNAPSHOT',
      knowledgeSource: 'kieStaging',
      process: 'evaluation@evaluation_1.0.0-SNAPSHOT',
    },
  },
  metaData: { status: 'draft' },
  errors: [],
};

const completionFormConfigs = {
  payload: {
    code: 'phase_1_widgets_completion_form',
    config: {
      containerId: 'evaluation_1.0.0-SNAPSHOT',
      knowledgeSource: 'kieStaging',
      process: 'evaluation@evaluation_1.0.0-SNAPSHOT',
      settings:
        '{"uiSchema":"{\\n  \\"47078d21-7da5-4d3f-8355-0fcd78b09f39\\": {\\n    \\"reason\\": {\\n      \\"ui:widget\\": \\"textarea\\"\\n    },\\n    \\"performance\\": {\\n      \\"ui:widget\\": \\"updown\\"\\n    }\\n  }\\n}"}',
    },
  },
  metaData: { status: 'draft' },
  errors: [],
};

const SummaryCardConfigs = {
  payload: {
    code: 'summary_card',
    config: {
      containerId: 'evaluation_1.0.0-SNAPSHOT',
      knowledgeSource: 'kieStaging',
      process: 'evaluation@evaluation_1.0.0-SNAPSHOT',
      settings: '{"summaryId": "request"}',
    },
  },
  metaData: { status: 'draft' },
  errors: [],
};

export default {
  TASK_LIST: taskListConfigs,
  TASK_DETAILS: taskDetailsConfigs,
  COMPLETION_FORM: completionFormConfigs,
  SUMMARY_CARD: SummaryCardConfigs,
};
