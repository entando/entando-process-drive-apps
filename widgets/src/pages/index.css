body {
  background-color: #f3f3f4;
}

.pda-wrapper {
  height: 100vh;
  display: grid;

  grid-template-areas:
    'header header'
    'sidebar content'
    'footer footer';

  grid-template-columns: 178px auto;
  grid-template-rows: auto 1fr auto;
  grid-gap: 0px;
}

.pda-header {
  grid-area: header;
}

.pda-sidebar {
  grid-area: sidebar;
  background-color: #2f4050;
}

.pda-content {
  grid-area: content;
  overflow-y: auto;
  background-color: #f3f3f3;
}

.pda-footer {
  grid-area: footer;
}

header {
  display: block;
  background-color: #ffffff;
  box-shadow: 0 1px 1px #cccccc;
  padding: 15px 15px 10px 10px;
  box-sizing: border-box;
}

.header-container {
  display: flex;
  justify-content: space-between;
  flex-wrap: nowrap;
}

.header-container > .header-logo {
  align-self: center;
}

.header-container > .header-controls {
  display: flex;
  align-self: center;
  margin-right: 5px;
}

.frame-container {
  margin: 15px 0px;
}

.frame-container .white-bg > h1 {
  background-color: #fff;
  padding: 12px 11px;
}

.dashboard-title h1 {
  font-size: 24px;
}

.pda-navigation li {
  padding: 10px 0;
}

.pda-navigation a {
  color: #a2acbd;
  font-size: 13px;
  font-weight: bold;
}

@media (min-width: 1600px) {
  .container {
    width: 100% !important;
  }
}
