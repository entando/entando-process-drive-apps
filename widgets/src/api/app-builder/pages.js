import { METHODS, DOMAINS } from 'api/constants';
import WIDGETS_CONFIG from 'mocks/app-builder/widgets';
import WIDGET_CONFIG_MOCKUPS from 'mocks/app-builder/pages';
import makeRequest from 'api/makeRequest';

// used to fetch mock response using frameId, it depends on WIDGETS_CONFIG setup
const getWidgetConfigMockResponse = frameId => {
  const widgetType = Object.keys(WIDGETS_CONFIG).find(
    iteratedType => WIDGETS_CONFIG[iteratedType].frameId === frameId
  );
  return WIDGET_CONFIG_MOCKUPS[widgetType];
};

export const getPageWidget = (pageCode, frameId) =>
  makeRequest({
    domain: DOMAINS.APP_BUILDER,
    uri: `/api/pages/${pageCode}/widgets/${frameId}`,
    method: METHODS.GET,
    mockResponse: getWidgetConfigMockResponse(frameId) || {},
    useAuthentication: true,
    // forceMock: true,
  });

export const putPageWidget = (pageCode, frameId, configs) =>
  makeRequest({
    domain: DOMAINS.APP_BUILDER,
    uri: `/api/pages/${pageCode}/widgets/${frameId}`,
    method: METHODS.PUT,
    body: configs,
    useAuthentication: true,
  });

// TODO: for ease of development - remove when authetication token is managed by wrapper
export const authenticate = async () => {
  const formData = {
    username: 'admin',
    password: 'adminadmin',
    grant_type: 'password',
  };
  return makeRequest({
    domain: DOMAINS.APP_BUILDER,
    uri: '/api/oauth/token',
    method: METHODS.POST,
    mockResponse: {
      access_token: 'MOCKED_ACCESS_TOKEN',
      token_type: 'bearer',
      refresh_token: 'MOCKED_REFRESH_TOKEN',
      expires_in: 3599,
    },
    body: Object.keys(formData)
      .map(key => `${encodeURIComponent(key)}=${encodeURIComponent(formData[key])}`)
      .join('&'),
    useAuthentication: false,
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded',
      Authorization: `Basic ${btoa('appbuilder:appbuilder_secret')}`,
    },
  });
};
