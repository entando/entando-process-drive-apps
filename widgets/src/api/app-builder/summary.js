import { SERVICE, METHODS } from 'api/constants';
import mockedSummary from 'mocks/summaries/summary';

import makeRequest from 'api/makeRequest';

export const getSummary = async (connection, summaryId, params = '?frequency=monthly') => {
  return makeRequest({
    domain: SERVICE.URL,
    uri: `/connections/${connection}/summaries/${summaryId}${params}`,
    method: METHODS.GET,
    mockResponse: mockedSummary,
    useAuthentication: false,
  });
};
