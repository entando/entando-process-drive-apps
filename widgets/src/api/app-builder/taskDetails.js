import { SERVICE, METHODS } from 'api/constants';
import mockedTask from 'mocks/taskDetails/task';
import mockedTaskForm from 'mocks/taskCompletionForm/formSchema';

import makeRequest from 'api/makeRequest';

export const getTask = async (connection, taskId) => {
  return makeRequest({
    domain: SERVICE.URL,
    uri: `/connections/${connection}/tasks/${taskId}`,
    method: METHODS.GET,
    mockResponse: mockedTask,
    useAuthentication: false,
  });
};

export const getTaskForm = async (connection, taskId) => {
  return makeRequest({
    domain: SERVICE.URL,
    uri: `/connections/${connection}/tasks/${taskId}/form`,
    method: METHODS.GET,
    mockResponse: mockedTaskForm,
    useAuthentication: false,
  });
};
