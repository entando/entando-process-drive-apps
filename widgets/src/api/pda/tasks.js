import { METHODS, DOMAINS } from 'api/constants';
import getMockedTasks from 'mocks/pda/taskList.api';

import makeRequest from 'api/makeRequest';

import COLUMNS from 'mocks/pda/columns';
import MOCKED_GET_TASK_RESPONSE from 'mocks/taskDetails/getTask';
import MOCKED_GET_TASK_FORM_RESPONSE from 'mocks/taskCompletionForm/getFormSchema';
import MOCKED_POST_TASK_FORM_RESPONSE from 'mocks/taskCompletionForm/postFormData';
import MOCKED_BULK_ACTION_RESPONSE from 'mocks/pda/bulkActions';

export const getTasks = async (
  { connection, groups },
  page = 0,
  pageSize = 30,
  sortedColumn,
  sortOrder,
  filter
) =>
  makeRequest({
    domain: DOMAINS.PDA,
    uri: `/connections/${connection}/tasks`,
    queryParams: {
      page: page + 1,
      pageSize,
      sort: sortedColumn,
      direction: sortOrder,
      filter,
      groups,
    },
    method: METHODS.GET,
    mockResponse: getMockedTasks(page, pageSize, sortedColumn, sortOrder, filter),
    useAuthentication: true,
  });

export const getColumns = async connection =>
  makeRequest({
    domain: DOMAINS.PDA,
    uri: `/connections/${connection}/tasks/columns`,
    method: METHODS.GET,
    mockResponse: COLUMNS,
    useAuthentication: true,
  });

export const getTask = async (connection, taskId) => {
  return makeRequest({
    domain: DOMAINS.PDA,
    uri: `/connections/${connection}/tasks/${taskId}`,
    method: METHODS.GET,
    mockResponse: MOCKED_GET_TASK_RESPONSE,
    useAuthentication: true,
  });
};

export const getTaskForm = async (connection, taskId) => {
  return makeRequest({
    domain: DOMAINS.PDA,
    uri: `/connections/${connection}/tasks/${taskId}/form`,
    method: METHODS.GET,
    useAuthentication: true,
    mockResponse: MOCKED_GET_TASK_FORM_RESPONSE.ALL_FIELDS,
  });
};

export const postTaskForm = async (connection, taskId, body) =>
  makeRequest({
    domain: DOMAINS.PDA,
    uri: `/connections/${connection}/tasks/${taskId}/form`,
    method: METHODS.POST,
    body: JSON.stringify(body),
    mockResponse: MOCKED_POST_TASK_FORM_RESPONSE,
    useAuthentication: true,
  });

export const TASK_BULK_ACTIONS = [
  'assign',
  'claim',
  'complete',
  'pause',
  'resume',
  'start',
  'unclaim',
];

export const putTasksBulkAction = async (connection, action, taskIds, assignee) => {
  const subUri = action === 'assign' && assignee ? `/${assignee}` : '';
  const uri = `/connections/${connection}/bulk/tasks/${action}${subUri}`;
  return makeRequest({
    domain: DOMAINS.PDA,
    uri,
    method: METHODS.PUT,
    body: JSON.stringify(taskIds),
    mockResponse: MOCKED_BULK_ACTION_RESPONSE,
    useAuthentication: true,
  });
};
