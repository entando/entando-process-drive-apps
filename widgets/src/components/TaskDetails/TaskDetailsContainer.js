import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from '@material-ui/core/styles';
import { Container, Box } from '@material-ui/core';

import { getTask } from 'api/pda/tasks';
import { getPageWidget } from 'api/app-builder/pages';
import theme from 'theme';
import CustomEventContext from 'components/TaskDetails/CustomEventContext';
import Overview from 'components/TaskDetails/Overview';
import GeneralInformation from 'components/TaskDetails/GeneralInformation';

const createLink = (pageCode = 'pda_task_details', taskId, locale = 'en') =>
  `/entando-de-app/${locale}/${pageCode}.page?taskId=${taskId}`;

class TaskDetailsContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      config: null,
      loadingTask: false,
      task: null,
    };

    this.fetchTask = this.fetchTask.bind(this);
    this.fetchWidgetConfigs = this.fetchWidgetConfigs.bind(this);
  }

  componentDidMount() {
    this.setState({ loadingTask: true }, async () => {
      const { config: storedConfig } = this.state;
      const config = storedConfig || (await this.fetchWidgetConfigs());

      this.setState({ config }, () => this.fetchTask());
    });
  }

  componentDidUpdate = prevProps => {
    const { taskId } = this.props;
    if (prevProps.taskId !== taskId) {
      this.fetchTask();
    }
  };

  async fetchWidgetConfigs() {
    const { pageCode, frameId } = this.props;
    try {
      // config will be fetched from app-builder
      const widgetConfigs = await getPageWidget(pageCode, frameId, 'TASK_DETAILS');
      if (widgetConfigs.errors && widgetConfigs.errors.length) {
        throw widgetConfigs.errors[0];
      }

      const { config } = widgetConfigs.payload;

      return config;
    } catch (error) {
      this.handleError(error.message);
    }
    return null;
  }

  async fetchTask() {
    const { config, loadingTask } = this.state;
    const { taskId } = this.props;

    const connection = (config && config.knowledgeSource) || '';

    if (!loadingTask) {
      this.setState({ loadingTask: true });
    }

    try {
      const task = await getTask(connection, taskId);

      this.setState({
        task: (task && task.payload) || null,
        taskInputData: (task && task.payload && task.payload.inputData) || {},
      });
    } catch (error) {
      this.handleError(error.message);
    } finally {
      this.setState({ loadingTask: false });
    }
  }

  handleError(err) {
    const { onError } = this.props;
    onError(err);
  }

  render() {
    const { loadingTask, task, taskInputData, config } = this.state;
    const { onPressPrevious, onPressNext, onError, taskId } = this.props;
    const configs = config && config.settings;
    console.log(configs);

    return (
      <CustomEventContext.Provider value={{ onPressPrevious, onPressNext, onError }}>
        <ThemeProvider theme={theme}>
          <Container disableGutters>
            <Box mb="20px">
              <Overview
                task={task}
                loadingTask={loadingTask}
                headerLabel={configs && configs.header}
                taskLink={createLink(configs && configs.destinationPageCode, taskId)}
              />
            </Box>
            {configs && configs.hasGeneralInformation && (
              <GeneralInformation taskInputData={taskInputData} loadingTask={loadingTask} />
            )}
          </Container>
        </ThemeProvider>
      </CustomEventContext.Provider>
    );
  }
}

TaskDetailsContainer.propTypes = {
  taskId: PropTypes.string.isRequired,
  onError: PropTypes.func,
  onPressPrevious: PropTypes.func,
  onPressNext: PropTypes.func,
  pageCode: PropTypes.string,
  frameId: PropTypes.string,
};

TaskDetailsContainer.defaultProps = {
  onError: () => {},
  onPressPrevious: () => {},
  onPressNext: () => {},
  pageCode: '',
  frameId: '',
};

export default TaskDetailsContainer;
