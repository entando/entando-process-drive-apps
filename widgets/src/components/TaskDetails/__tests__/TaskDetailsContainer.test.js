import { render, wait } from '@testing-library/react';
import React from 'react';
import 'mocks/i18nMock';

import WIDGET_CONFIGS from 'mocks/app-builder/widgets';
import MOCKED_GET_TASK_RESPONSE from 'mocks/taskDetails/getTask';
import { getTask } from 'api/pda/tasks';
import { getPageWidget } from 'api/app-builder/pages';

import TaskDetailsContainer from 'components/TaskDetails/TaskDetailsContainer';

jest.mock('api/app-builder/pages');
jest.mock('api/pda/tasks');

beforeEach(() => {
  getPageWidget.mockClear();
  getTask.mockClear();
});

describe('<TaskDetailsContainer />', () => {
  getPageWidget.mockImplementation(() => Promise.resolve(WIDGET_CONFIGS.TASK_DETAILS.configs));
  getTask.mockImplementation(() => Promise.resolve(MOCKED_GET_TASK_RESPONSE));

  it('renders snapshot correctly', async () => {
    const { container } = render(
      <TaskDetailsContainer
        taskId={WIDGET_CONFIGS.TASK_DETAILS.taskId}
        pageCode={WIDGET_CONFIGS.TASK_DETAILS.pageCode}
        frameId={WIDGET_CONFIGS.TASK_DETAILS.frameId}
        widgetCode={WIDGET_CONFIGS.TASK_DETAILS.widgetCode}
      />
    );

    await wait(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
