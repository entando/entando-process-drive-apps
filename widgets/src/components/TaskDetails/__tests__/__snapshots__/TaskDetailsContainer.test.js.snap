// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskDetailsContainer /> renders snapshot correctly 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthLg"
  >
    <div
      class="MuiPaper-root WidgetBox-widgetBox-10 TaskDetailsContainer-taskDetailWidgetBox-1 MuiPaper-outlined"
    >
      <div>
        <div
          class="MuiBox-root MuiBox-root-193"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2"
          >
            taskDetails.overview.title - 293
          </h2>
          <div
            aria-label="outlined primary button group"
            class="MuiButtonGroup-root"
            role="group"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-outlinedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                    />
                  </svg>
                </span>
                taskDetails.overview.previous
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-outlinedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                taskDetails.overview.next
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    />
                  </svg>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <label
              class="MuiFormLabel-root"
            >
              task.fields.createdBy
            </label>
            <div
              class="OverviewDetails-value-288"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                pamAdmin
              </p>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <label
              class="MuiFormLabel-root"
            >
              task.fields.name
            </label>
            <div
              class="OverviewDetails-value-288"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                PM Evaluation
              </p>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <label
              class="MuiFormLabel-root"
            >
              task.fields.createdOn
            </label>
            <div
              class="OverviewDetails-value-288"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                11/14/2019, 11:26:30 AM
              </p>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <label
              class="MuiFormLabel-root"
            >
              task.fields.status
            </label>
            <div
              class="OverviewDetails-value-288"
            >
              <div
                class="MuiChip-root BadgeChip-root-297"
              >
                <span
                  class="MuiChip-label"
                >
                  CREATED
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <label
              class="MuiFormLabel-root"
            >
              task.fields.due
            </label>
            <div
              class="OverviewDetails-value-288"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                -
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root WidgetBox-widgetBox-10 MuiPaper-outlined"
    >
      <div>
        <h3
          class="MuiTypography-root MuiTypography-h3"
        >
          taskDetails.generalInformation.title
        </h3>
        <hr
          class="MuiDivider-root GeneralInformation-divider-39"
        />
        <div
          class="GeneralInformation-taskDatum-40"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <span
              class="GeneralInformation-label-41"
            >
              task.inputData.Comment
              :
            </span>
            You need to evaluate Alessio.
          </p>
        </div>
        <div
          class="GeneralInformation-taskDatum-40"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <span
              class="GeneralInformation-label-41"
            >
              task.inputData.reason
              :
            </span>
            asasd
          </p>
        </div>
        <div
          class="GeneralInformation-taskDatum-40"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <span
              class="GeneralInformation-label-41"
            >
              task.inputData.performance
              :
            </span>
            12
          </p>
        </div>
        <div
          class="GeneralInformation-taskDatum-40"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <span
              class="GeneralInformation-label-41"
            >
              task.inputData.TaskName
              :
            </span>
            PerformanceEvaluation
          </p>
        </div>
        <div
          class="GeneralInformation-taskDatum-40"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <span
              class="GeneralInformation-label-41"
            >
              task.inputData.NodeName
              :
            </span>
            PM Evaluation
          </p>
        </div>
        <div
          class="GeneralInformation-taskDatum-40"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <span
              class="GeneralInformation-label-41"
            >
              task.inputData.Skippable
              :
            </span>
            false
          </p>
        </div>
        <div
          class="GeneralInformation-taskDatum-40"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <span
              class="GeneralInformation-label-41"
            >
              task.inputData.ActorId
              :
            </span>
            pamAdmin
          </p>
        </div>
        <div
          class="GeneralInformation-taskDatum-40"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <span
              class="GeneralInformation-label-41"
            >
              task.inputData.BusinessAdministratorId
              :
            </span>
            pamAdmin
          </p>
        </div>
        <div
          class="GeneralInformation-taskDatum-40"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <span
              class="GeneralInformation-label-41"
            >
              task.inputData.GroupId
              :
            </span>
            PM
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
